qcodes/tests/test_parameter.py

```diff
@@ -2052,18 +2073,332 @@ def __init__(self, *args, **kwargs):
-        local_parameter.cache.get()
+    if get_if_invalid is True:
+        with pytest.raises(RuntimeError, match="max_val_age` is not supported"):
+            local_parameter.cache.get(get_if_invalid=get_if_invalid)
+    elif get_if_invalid == NOT_PASSED:
+        with pytest.raises(RuntimeError, match="max_val_age` is not supported"):
+            local_parameter.cache.get()
+    else:
+        assert local_parameter.cache.get(get_if_invalid=get_if_invalid) == 1
 
 
-def test_no_get_timestamp_none_runtime_error():
+def test_no_get_timestamp_none_runtime_error(get_if_invalid):
     """
     Test that a parameter that has never been
-    set and does not support getting raises a RuntimeError
-    when trying to get the cache.
+    set, cannot be get and does not support
+    getting raises a RuntimeError.
     """
     local_parameter = Parameter('test_param', get_cmd=False)
 
-    with pytest.raises(RuntimeError, match="Value of parameter test_param"):
-        local_parameter.cache.get()
+    if get_if_invalid is True:
+        with pytest.raises(RuntimeError, match="Value of parameter test_param"):
+            local_parameter.cache.get(get_if_invalid=get_if_invalid)
+    elif get_if_invalid == NOT_PASSED:
+        with pytest.raises(RuntimeError, match="Value of parameter test_param"):
+            local_parameter.cache.get()
+    else:
+        assert local_parameter.cache.get(get_if_invalid=get_if_invalid) is None
+
+
+NOT_PASSED = 'NOT_PASSED'
+
+
+@pytest.fixture(params=(True, False, NOT_PASSED))
+def snapshot_get(request):
+    return request.param
+
+
+@pytest.fixture(params=(True, False, NOT_PASSED))
+def snapshot_value(request):
+    return request.param
+
+
+@pytest.fixture(params=(None, False, NOT_PASSED))
+def get_cmd(request):
+    return request.param
+
+
+@pytest.fixture(params=(True, False, NOT_PASSED))
+def get_if_invalid(request):
+    return request.param
+
+
+def create_parameter(snapshot_get, snapshot_value, cache_is_valid, get_cmd,
+                     offset=NOT_PASSED):
+    kwargs = dict(set_cmd=None,
+                  label='Parameter',
+                  unit='a.u.',
+                  docstring='some docs')
+
+    if offset != NOT_PASSED:
+        kwargs.update(offset=offset)
+
+    if snapshot_get != NOT_PASSED:
+        kwargs.update(snapshot_get=snapshot_get)
+
+    if snapshot_value != NOT_PASSED:
+        kwargs.update(snapshot_value=snapshot_value)
+
+    if get_cmd != NOT_PASSED:
+        kwargs.update(get_cmd=get_cmd)
+
+    p = Parameter('p', **kwargs)
+
+    if get_cmd is not False:
+        def wrap_in_call_counter(get_func):
+            call_count = 0
+
+            def wrapped_func(*args, **kwargs):
+                nonlocal call_count
+                call_count += 1
+                return get_func(*args, **kwargs)
+
+            wrapped_func.call_count = lambda: call_count
+
+            return wrapped_func
+
+        p.get = wrap_in_call_counter(p.get)
```

Comment:
ooops, overwriting `get`...

Reply:
Yes but does overwrite it with a wrapped version of it self so this is safe as is